### PR TITLE
Handle redirect for ChromeTab Auth Code flow.

### LINKED ISSUE #159: Auth Code flow via ChromeTab fails
The redirect from the Auth Code flow via ChromeTab fails to parse the auth code from the redirect uri.  This the redirect currently works for web view flow where a redirect handler is used but not in the ChromeTab fallback implementation.

Library version:  0.9.x, 0.10.x

### PR DESCRIPTION
**Pull request by thepoofy**
fixes #159

Description:  SDK is expected to pass Auth Code to app and app developer is expected to use their own service to exchange auth code for access token.  

Related issue(s): #159

### CODE REVIEW

File: core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java [hunk 1]
```diff
@@ -47,7 +44,7 @@
     private static final String BEARER = "Bearer";
 
     private final String ACCESS_TOKEN_STRING = "accessToken1234";
-    private final long EXPIRATION_TIME = 1458770906206l;
+    private final long EXPIRATION_TIME = 1458770906206L;
```
Comment (ZacSweers) [inline core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java]: nit: comment with what this is in human terms?

File: core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java [hunk 2]
```diff
@@ -237,25 +234,36 @@ public void generateAccessTokenFromUrl_whenValidAccessTokenWithMultipleScopes_sh
...
+
     @Test
     public void getCodeFromUrl_whenValidAuthorizationCodePassed() throws LoginAuthenticationException {
         String redirectUrl = "http://localhost:1234?code=" + AUTH_CODE;
 
         assertThat(AuthUtils.parseAuthorizationCode(Uri.parse(redirectUrl))).isEqualTo(AUTH_CODE);
     }
 
-    @Test
+    @Test(expected = LoginAuthenticationException.class)
```
Comment (ZacSweers) [inline core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java]: nit: I like to assert something about this, such as a partial message

File: core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java (hunk 1 above)
Comment (thepoofy) [reply core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java]: Our linter prefers using an uppercase "L" over lowercase, probably for readability.

File: core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java (hunk 2 above)
Comment (thepoofy) [reply core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java]: Would you prefer a try/catch in the test as the prior implementation used?

Comment (ZacSweers) [reply core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java]: yeah. Good/bad example https://github.com/uber/AutoDispose/blob/6e1399512f132bc5052c4b1ba111debb6eb430e1/android/autodispose-android/src/test/java/com/uber/autodispose/android/AutoDisposeAndroidPluginsTest.java#L56

File: core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java (hunk 1 above)
Comment (ZacSweers) [reply core-android/src/test/java/com/uber/sdk/android/core/auth/AuthUtilsTest.java]: I meant like, `// august 10th, 48196`

**Comment by jck2** [comment]
Are we sure this issue description is true? I'm pretty sure the auth code callback here (`onCodeReceived`) is actually deprecated. Probably worth double checking, but I'm fairly certain that the https url that gets registered in the sessionConfiguration gets hit directly by the server and the client isn't responsible for doing that part of the code exchange.

**Comment by tyvsmith** [comment]
What's the behavior now? Previously, a developer would expect the first redirect to be to their owned server to parse out code, followed by the second redirect back to their own App URI.

**Comment by jck2** [comment]
@tyvsmith My understanding was that what you said would be the intended behavior. In that case though there's no reason to send the Auth Code to the calling app in onCodeReceived, we should only need to return the access token/refresh token.

**Comment by thepoofy** [comment]
The SDK can currently be configured as a callback for an auth code flow.  When this flow is done on a device that supports ChromeTabs the redirect back to the SDK will fail every time as we only have support for WebView handling of this.  I previously closed this PR based upon desire to move away from allowing an SDK to handle an Auth Code redirect but I believe there is a legit use case for it.

If the end user wishes to use their app to manually exchange auth code for an access token they should be allowed to.  It's probably not safe to do in a production environment but for fast prototyping and development this is valuable.  eg: Creating an activity with an intent filter for a https url which will be be built at a later date or using an secured API to exchange their code for a token.

This use of the auth code flow is still marked deprecated.

**Comment by jck2** [comment]
I think we have some features that are only enabled in debug mode. I think that's a valid use case but I would prefer that we don't allow it in production.

We have a `Utility.isDebuggable` method that should enable this gating

**Pull request merged**